### ISSUE TITLE
Indentation of variables

### ISSUE THREAD

**Issue by ArmorDarks** [brigade/scss-lint#618]
Options:

``` yaml
  Indentation:
    enabled: true
    allow_non_nested_indentation: true
    character: tab
    width: 1
```

When `allow_non_nested_indentation` enabled I thinks it's naturally to expect that you could intend variables as you might need too.

But this example

``` scss
.site-header {}

	.site-logo {}

		.site-logo__link {}

	.site-nav {}

		$site-nav__item-color: #000;

		.site-nav__item {

			color: $site-nav__item-color;

		}
```

will throw

```
9: [W] Indentation:  Line should be indented 0 tabs, but was indented 2 tabs
```

I agree that it isn't best example, since you can put variable inside of selector's scope and it will fix that error. But imagine that you need to keep it out of selectors scope, to make it available to other selectors below, but in same time you want to keep that variable close to it's origin.

This definitely looks wrong (but acceptable for SCSS Lint):

``` scss
.site-header {}

	.site-logo {}

		.site-logo__link {}

	.site-nav {}

$site-nav__item-color: #000;

		.site-nav__item {

			color: $site-nav__item-color;

		}
```

I must admit that for better flexibility it would be better to assign different option for controlling variable's indentation

**Comment by srawlins** [comment]
@sds Here's the code in question:

```ruby
def check_arbitrary_indent(node, actual_indent)
  return if check_root_ruleset_indent(node, actual_indent)

  if @indent == 0
    unless node.is_a?(Sass::Tree::RuleNode) || actual_indent == 0
      add_lint(node.line, lint_message(0, actual_indent))
      return true
    end
    # ...
```

It seems like the general fix would just be to never lint `if @indent == 0`?Or do we want a long list of valid nodes in that `unless` condition? Variables, IfNodes, ForNodes, etc?

**Comment by sds** [comment]
@srawlins @ArmorDarks Interestingly enough, we have a spec that explicitly considers this case an offense: https://github.com/brigade/scss-lint/blob/v0.42.2/spec/scss_lint/linter/indentation_spec.rb#L327-L334

However, I couldn't tell you why it's considered an offense. :)

I'm open to adding more nodes to the whitelisted set.

**Comment by sds** [comment]
Nevermind, changing this to allow any node that is a direct child of the document root to have arbitrary indent makes sense here.

Thanks doing the investigation, @srawlins. I've implemented a fix in 67c5fe5b224489dd7a8df6ccbc4400e0deafc285.
